services/blockchain-indexer/shared/utils/snapshot.js

```diff
@@ -0,0 +1,142 @@
+const https = require('https');
+const zlib = require('zlib');
+const util = require('util');
+const config = require('../../config');
+const execInShell = util.promisify(require('child_process').exec);
+
+const { mkdir, exists } = require('./fs');
+const { requestConnector } = require('./request');
+
+const logger = Logger();
+
+let snapshotFilePath = './data/service-snapshot.sql';
+
+const getHTTPProtocolByURL = (url) => url.startsWith('https') ? https : http;
+
+const checkCommandAvailability = async () => {
+	const { stdout: mysqlAvailable } = await execInShell('which mysql').catch(() => ({}));
+	if (!mysqlAvailable) throw new NotFoundException('mysql command is unavailable in PATH');
+};
+
+function downloadAndUnzipFile(fileUrl, filePath) {
+	return new Promise((resolve, reject) => {
+		getHTTPProtocolByURL(fileUrl).get(fileUrl, (response) => {
+			if (response.statusCode === 200) {
+				const unzip = zlib.createUnzip();
+				const writeFile = fs.createWriteStream(filePath);
+
+				response.pipe(unzip).pipe(writeFile);
+
+				response.on('error', (err) => {
+					reject(new Error(err));
+				});
+
+				writeFile.on('finish', () => {
+					resolve();
+				});
+
+				writeFile.on('error', (err) => {
+					reject(new Error(err));
+				});
+			} else {
+				const errMessage = `Download failed with HTTP status code: ${response.statusCode} (${response.statusMessage})`;
+				console.error(errMessage);
+				if (response.statusCode === 404) {
+					reject(new Error(`NotFoundException: ${errMessage}`));
+				} else {
+					reject(new Error(errMessage));
+				}
+			}
+		}).on('error', (err) => {
+			reject(new Error(err));
+		});
+	});
+}
+
+const resolveSnapshotRestoreCommand = async (connEndpoint) => {
+	await checkCommandAvailability();
+	const [user, password] = connEndpoint.split('//')[1].split('@')[0].split(':');
+	const [host, port, database] = connEndpoint.split('@')[1].split(new RegExp('/|:', 'g'));
+	const mysqlSnapshotCommand = `mysql ${database} -h ${host} -P ${port} -u ${user} -p${password} < ${snapshotFilePath}`;
+	return mysqlSnapshotCommand;
+};
+
+const applySnapshot = async (connEndpoint = config.endpoints.mysql) => {
+	try {
+		logger.debug('Attempting to resolve the snapshot command.');
+		const snapshotRestoreCommand = await resolveSnapshotRestoreCommand(connEndpoint);
+		logger.info(`Attempting to apply the snapshot file available at: ${snapshotFilePath}.`);
+		const { stdout, stderr } = await execInShell(snapshotRestoreCommand);
+		logger.info(stdout);
+		logger.warn(stderr);
+		logger.info('SQL file(s) imported successfully.');
+	} catch (error) {
+		logger.error(error.stack);
+		throw error;
+	}
+};
+
+const downloadSnapshot = async (snapshotUrl) => {
+	const directoryPath = path.dirname(snapshotFilePath);
+	if (!(await exists(directoryPath))) await mkdir(directoryPath, { recursive: true });
+	logger.info('Attempting to download the snapshot file.');
+	await downloadAndUnzipFile(snapshotUrl, snapshotFilePath);
```

Comment:
Please add the checksum verification logic for the downloaded snapshots. For the (default) configured snapshot URLs in config.js, the checksum file will always be `${snapshotURL.replace('gz', 'SHA256')}`. Eg: `https://snapshots.lisk.com/mainnet/service.sql.SHA256` for mainnet.